### PR TITLE
fix: Make copied event sponsor logo url generation a celery task

### LINKED ISSUE #5866: Sponsored events with images not getting copied
**Describe the bug**

The image resizing task needs to be turned into a celery task to avoid server block

**To Reproduce**
1. Try copying an event with Sponsor images
2. See error

**Expected behavior**

The `create_save_resized_image` function shouldn't jam the server

**Additional details (please complete the following information):**
 - OS: [e.g. MacOS, Ubuntu, CentOS] Ubuntu 16.04 LTS
 - Python Version [e.g. `3.5`, `3.6`] 3.6

**Additional context**

Looking into it

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #5866 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:

- creates the task along the lines of event image resizing tasks

### CODE REVIEW

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review! :)

File: app/api/helpers/tasks.py [hunk 1]
```diff
@@ -120,6 +121,19 @@ def resize_user_images_task(self, user_id, original_image_url):
         logging.exception('Error encountered while generating resized images for user with id: {}'.format(user_id))
 
 
+@celery.task(base=RequestContextTask, name='sponsor.logo.urls', bind=True)
+def sponsor_logos_url_task(self, event_id):
+    sponsor = Sponsor.query.filter_by(event_id=event_id).first()
```
Comment (CosmicCoder96) [inline app/api/helpers/tasks.py]: This will just do it for the first sponsor?

Comment (uds5501) [reply app/api/helpers/tasks.py]: Not really, since the function is called for each sponsor, every sponsor is taken care of @CosmicCoder96

Comment (iamareebjamal) [reply app/api/helpers/tasks.py]: Even for each sponsor, the same sponsor is being fetched everytime

Comment (uds5501) [reply app/api/helpers/tasks.py]: Oh, I see the point now, will apply necessary changes and push the PR ASAP @iamareebjamal @CosmicCoder96

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Please review now

File: app/api/helpers/tasks.py [hunk 2]
```diff
@@ -120,6 +121,20 @@ def resize_user_images_task(self, user_id, original_image_url):
         logging.exception('Error encountered while generating resized images for user with id: {}'.format(user_id))
 
 
+@celery.task(base=RequestContextTask, name='sponsor.logo.urls', bind=True)
+def sponsor_logos_url_task(self, event_id):
+    sponsors = Sponsor.query.filter_by(event_id=event_id).all()
```
Comment (iamareebjamal) [inline app/api/helpers/tasks.py]: is_active=True

Comment (uds5501) [reply app/api/helpers/tasks.py]: @iamareebjamal Didn't get it :sweat_smile:

Comment (uds5501) [reply app/api/helpers/tasks.py]: @iamareebjamal  Did you mean it like this?

Comment (iamareebjamal) [reply app/api/helpers/tasks.py]: You should only get sponsors where is_active is True, getting all means you'll get deleted sponsors as well

**Comment by uds5501** [comment]
@iamareebjamal @CosmicCoder96 Have a look now?

**Pull request merged**